// @storybook/addon-queryparams/dist/esm/index.js
import "core-js/modules/es.array.iterator.js";
import "core-js/modules/es.object.assign.js";
import "core-js/modules/es.object.to-string.js";
import "core-js/modules/es.regexp.exec.js";
import "core-js/modules/es.regexp.to-string.js";
import "core-js/modules/es.string.iterator.js";
import "core-js/modules/es.string.search.js";
import "core-js/modules/web.dom-collections.iterator.js";
import "core-js/modules/web.url.js";
import { document, history } from 'global';
import qs from 'qs';
import { makeDecorator } from '@storybook/addons';
import { PARAM_KEY } from './constants';
export var withQuery = makeDecorator({
  name: 'withQuery',
  parameterName: PARAM_KEY,
  skipIfNoParametersOrOptions: true,
  wrapper: function wrapper(getStory, context, _ref) {
    var parameters = _ref.parameters;
    var location = document.location;
    var currentQuery = qs.parse(location.search, {
      ignoreQueryPrefix: true
    });
    var additionalQuery = typeof parameters === 'string' ? qs.parse(parameters, {
      ignoreQueryPrefix: true
    }) : parameters;
    var newLocation = new URL(document.location.href);
    newLocation.search = qs.stringify(Object.assign({}, currentQuery, additionalQuery));
    history.replaceState({}, document.title, newLocation.toString());
    return getStory(context);
  }
});

if (module && module.hot && module.hot.decline) {
  module.hot.decline();
}
